### ISSUE TITLE
throw-error: allow eventEmitter.on('error', reject)

### ISSUE THREAD

**Issue by felixfbecker** [cartant/eslint-plugin-etc#1]
A pattern I commonly have is rejecting a Promise when a Node event emitter emits an `error` event. These are usually `Error`, but really on the type level `any`. Therefor I would expect `throw-error` to allow `.on('error', reject)` as `any` is assignable to `Error`.

Example:
```ts
export const watchIntegrationAssets = gulp.series(
    copyIntegrationAssets,
    async function watchIntegrationAssets(): Promise<void> {
        await new Promise<never>((_, reject) => {
            gulp.watch(INTEGRATION_FILES, copyIntegrationAssets).on('error', reject)
        })
    }
)
```

**Comment by cartant** [comment]
That shouldn't be a problem. Just curious, did the TSLint rule support that usage? The implementation of the ESLint rule is very different.

**Comment by felixfbecker** [comment]
I don't remember it erroring on this one, so I assume so. Might also just be because it caught less cases in general

**Comment by cartant** [comment]
Should be fixed in `0.0.1-beta.7`.

**Comment by felixfbecker** [comment]
Hmm, I upgraded and still get the same error with the above code snippet

**Comment by cartant** [comment]
That's weird. The test with your snippet, from above, passes.

**Comment by cartant** [comment]
Refactored it to test `callee` instead of `parameters` - which should make no difference to the implementation. Re-published it as `0.0.1-beta.8` in case I mess something up when I made the first change (in the airport). If I comment out [the change](https://github.com/cartant/eslint-plugin-etc/blob/7bb19b3c89fe32a07b6f8a0a5c7f6aa82991db7d/source/rules/throw-error.ts#L77), the test fails, so ... 🤷‍♂

**Comment by cartant** [comment]
Closing this because ... 🤷‍♂ ... because it needs a failing snippet/test for something to be done about it.
